flare/framework/tests/test_recurrent.py

```diff
@@ -0,0 +1,115 @@
+import flare.framework.recurrent as rc
+import unittest
+import numpy as np
+import torch
+
+
+def tensor_lists_equal(t1, t2):
+    """
+    Given two (nested) lists of tensors, return whether they are equal.
+    """
+    if isinstance(t1, torch.Tensor) and isinstance(t2, torch.Tensor):
+        return bool((t1 == t2).all().item())
+
+    assert isinstance(t1, list)
+    assert isinstance(t2, list)
+    assert len(t1) == len(t2)
+
+    for t1_, t2_ in zip(t1, t2):
+        if not tensor_lists_equal(t1_, t2_):
+            return False
+    return True
+
+
+class TestHierarchyTensorTranspose(unittest.TestCase):
+    def test_transpose(self):
+        data = [[[[3], [4], [5]], [[1], [2]], [[10], [2], [4], [5]]],
+                [[[3], [4], [5]], [[1], [2]]]]
+
+        ht = rc.make_hierarchy_of_tensors(data, "int64", "cpu", (1, ))
+        ht_ = rc.transpose(rc.transpose(ht))  ## this should convert back to ht
```

Comment:
This test case might be improved? For example, if the transpose() function is implemented as an identity function, it can also pass this test case.

Reply:
But the transpose() is clearly not implemented as identity.